lib/Twig/Cache/Filesystem.php

```diff
@@ -58,8 +58,11 @@ public function write($key, $content)
     {
         $dir = dirname($key);
         if (!is_dir($dir)) {
-            if (false === @mkdir($dir, 0777, true) && !is_dir($dir)) {
-                throw new RuntimeException(sprintf('Unable to create the cache directory (%s).', $dir));
+            if (false === @mkdir($dir, 0777, true)) {
+                clearstatcache(false, $dir);
+                if (!is_dir($dir)) {
+                    throw new RuntimeException(sprintf('Unable to create the cache directory (%s).', $dir));
+                }
```

Comment:
mkdir()  will create directory & set directory permission atomically.
So no need to care race condition problem with is_writable() ?